packages/react-native/scripts/cocoapods/utils.rb

```diff
@@ -195,16 +195,21 @@ def self.apply_xcode_15_patch(installer, xcodebuild_manager: Xcodebuild)
     private
 
     def self.add_build_settings_to_pod(installer, settings_name, settings_value, target_pod_name, configuration_type)
+        # [macOS
+        # Since some RN projects might combine multiple platforms into the same Xcode project,
+        # we'll be much more forgiving with the pod name matching in react-native-macos.
+        valid_suffixes = ["", "-iOS", "-macOS", "-visionOS"]
```

Comment:
Can we add a GitHub issue here for tracking the upstream fix? It feels simple enough to make the case to expand the use of $RN_PLATFORMS here to construct these long term (I disagree that isn't what the array is used for, but we can figure that out in the upstream fix)